# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/gc/GC.cpp
# Commit: b29a261a06ec
# Full Hash: b29a261a06ec664f02c57121ce6e5302590fa7ac
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2022-03-04 04:18:28
# Regressor Bug: 1643586
# File Overlap Count: 1
# Description:
#   Bug 1643586 - Part 4: Remove flag for disabling compartment checks from JSContext r=jandem
#   
#   This is only used in one place and it's simpler to check the tracer kind there.
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D140191
# ==============================================================================

diff -r 619f2aabe4d5 -r b29a261a06ec js/src/gc/GC.cpp
--- a/js/src/gc/GC.cpp	Thu Mar 03 17:49:17 2022 +0000
+++ b/js/src/gc/GC.cpp	Thu Mar 03 17:49:17 2022 +0000
@@ -2137,7 +2137,7 @@
 
  public:
   explicit CompartmentCheckTracer(JSRuntime* rt)
-      : JS::CallbackTracer(rt, JS::TracerKind::Callback,
+      : JS::CallbackTracer(rt, JS::TracerKind::CompartmentCheck,
                            JS::WeakEdgeTraceAction::Skip),
         src(nullptr),
         zone(nullptr),
@@ -2175,9 +2175,7 @@
   Compartment* comp =
       MapGCThingTyped(thing, [](auto t) { return t->maybeCompartment(); });
   if (comp && compartment) {
-    if (!runtime()->mainContextFromOwnThread()->disableCompartmentCheckTracer) {
-      MOZ_ASSERT(comp == compartment || edgeIsInCrossCompartmentMap(thing));
-    }
+    MOZ_ASSERT(comp == compartment || edgeIsInCrossCompartmentMap(thing));
   } else {
     TenuredCell* tenured = &thing.asCell()->asTenured();
     Zone* thingZone = tenured->zoneFromAnyThread();